File: stripe/src/main/java/com/stripe/android/UidSupplier.java
package com.stripe.android;

import android.content.ContentResolver;
import android.content.Context;
import android.provider.Settings;
import android.support.annotation.NonNull;

import com.stripe.android.utils.ObjectUtils;

final class UidSupplier implements Supplier<StripeUid> {
    @NonNull private final ContentResolver mContentResolver;

    UidSupplier(@NonNull Context context) {
        mContentResolver = context.getApplicationContext().getContentResolver();
    }

    @SuppressWarnings("HardwareIds")
    @NonNull
    public StripeUid get() {
        final String androidId =
                Settings.Secure.getString(mContentResolver, Settings.Secure.ANDROID_ID);
        return StripeUid.create(ObjectUtils.getOrDefault(androidId, ""));
    }
}

File: stripe/src/main/java/com/stripe/android/UidSupplier.kt
package com.stripe.android

import android.annotation.SuppressLint
import android.content.ContentResolver
import android.content.Context
import android.provider.Settings

internal class UidSupplier(context: Context) : Supplier<StripeUid> {
    private val contentResolver: ContentResolver = context.applicationContext.contentResolver

    private val deviceId: String?
        @SuppressLint("HardwareIds")
        get() {
            return Settings.Secure.getString(contentResolver, Settings.Secure.ANDROID_ID)
        }

    @SuppressLint("HardwareIds")
    override fun get(): StripeUid {
        return StripeUid(deviceId.orEmpty())
    }
}
